onnxruntime/core/optimizer/conv_activation_fusion.cc

```diff
@@ -1,162 +1,286 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-#include <deque>
-#include "core/graph/graph_utils.h"
 #include "core/optimizer/conv_activation_fusion.h"
-#include "core/optimizer/initializer.h"
+
+#include <string_view>
+
+#include "core/common/inlined_containers.h"
+#include "core/framework/tensorprotoutils.h"
+#include "core/graph/graph_utils.h"
 #include "core/optimizer/utils.h"
+#include "core/optimizer/selectors_actions/actions.h"
 
-using namespace ONNX_NAMESPACE;
-using namespace ::onnxruntime::common;
 namespace onnxruntime {
 
-Status ConvActivationFusion::ApplyImpl(Graph& graph, bool& modified, int graph_level, const logging::Logger& logger) const {
-  GraphViewer graph_viewer(graph);
-  const auto& order = graph_viewer.GetNodesInTopologicalOrder();
+namespace {
 
-  for (auto index : order) {
-    auto* node = graph.GetNode(index);
-    // check that node hasn't already been removed
-    if (!node)
-      continue;
+#if !defined(ORT_MINIMAL_BUILD)
+namespace selectors {
+const Node* GetLoneConsumerNode(const GraphViewer& graph_viewer, const Node& node) {
+  if (graph_viewer.NodeProducesGraphOutput(node) || node.GetOutputEdgesCount() != 1) {
+    return nullptr;
+  }
+  return &*node.OutputNodesBegin();
+}
```

Comment:
Can optimizer_utils::CheckOutputEdges be used?

Reply:
would need to add a separate version that takes a GraphViewer

Reply:
Could we use GraphViewer::GetGraph and avoid adding a new variant?

Reply:
yes, also removed the GetClipConstantMinMax() variant